File: scripts/tools/generate_tray_icons.py
#!/usr/bin/env python3
"""
Generate tray icons for macOS menu bar
Creates simple, clean icons suitable for menu bar display
"""

from PIL import Image, ImageDraw
import os

# Output directory
output_dir = os.path.join(os.path.dirname(__file__), '..', 'src-tauri', 'resources')
os.makedirs(output_dir, exist_ok=True)

# Icon size for macOS menu bar (22x22 @ 2x = 44x44)
SIZE = 44
CENTER = SIZE // 2

def create_base_microphone(draw, color, opacity=255):
    """Draw a simple microphone icon"""
    # Microphone capsule
    mic_width = 12
    mic_height = 18
    mic_left = CENTER - mic_width // 2
    mic_top = CENTER - mic_height // 2 - 4
    mic_right = CENTER + mic_width // 2
    mic_bottom = mic_top + mic_height

    # Draw microphone body (rounded rectangle)
    draw.rounded_rectangle(
        [mic_left, mic_top, mic_right, mic_bottom],
        radius=6,
        fill=(*color, opacity)
    )

    # Microphone stand
    stand_y = mic_bottom + 2
    draw.line([CENTER, mic_bottom, CENTER, stand_y + 6], fill=(*color, opacity), width=2)

    # Base
    base_width = 10
    draw.line(
        [CENTER - base_width//2, stand_y + 6, CENTER + base_width//2, stand_y + 6],
        fill=(*color, opacity),
        width=2
    )

def create_sound_waves(draw, color, opacity=255, side='left', offset=0):
    """Draw sound waves on one side"""
    wave_opacity = int(opacity * 0.6)

    if side == 'left':
        x = CENTER - 12 - offset
        # Inner wave
        draw.arc([x-3, CENTER-5, x, CENTER+5], 270, 90, fill=(*color, wave_opacity), width=2)
        # Outer wave
        draw.arc([x-6, CENTER-8, x, CENTER+8], 270, 90, fill=(*color, int(wave_opacity*0.7)), width=2)
    else:  # right
        x = CENTER + 12 + offset
        # Inner wave
        draw.arc([x, CENTER-5, x+3, CENTER+5], 90, 270, fill=(*color, wave_opacity), width=2)
        # Outer wave
        draw.arc([x, CENTER-8, x+6, CENTER+8], 90, 270, fill=(*color, int(wave_opacity*0.7)), width=2)

def create_tray_icon(filename, state, is_dark=False):
    """
    Create a tray icon
    state: 'idle', 'recording', 'transcribing'
    is_dark: True for dark mode icons
    """
    # Create image with transparency
    img = Image.new('RGBA', (SIZE, SIZE), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)

    # Color scheme
    if is_dark:
        # For dark mode menu bar (light icons)
        base_color = (255, 255, 255)
    else:
        # For light mode menu bar (dark icons)
        base_color = (0, 0, 0)

    if state == 'idle':
        # Simple microphone, no waves
        create_base_microphone(draw, base_color, opacity=200)

    elif state == 'recording':
        # Microphone with red tint and waves
        if is_dark:
            mic_color = (255, 100, 100)  # Light red for dark mode
        else:
            mic_color = (220, 50, 50)  # Dark red for light mode

        create_base_microphone(draw, mic_color, opacity=255)
        create_sound_waves(draw, mic_color, opacity=255, side='left')
        create_sound_waves(draw, mic_color, opacity=255, side='right')

        # Add a small recording indicator (dot)
        dot_radius = 2
        draw.ellipse(
            [CENTER - dot_radius, mic_top - 6, CENTER + dot_radius, mic_top - 2],
            fill=(255, 50, 50, 255)
        )

    elif state == 'transcribing':
        # Microphone with blue/purple tint and animated waves
        if is_dark:
            mic_color = (167, 139, 250)  # Light purple for dark mode
        else:
            mic_color = (139, 92, 246)  # Darker purple for light mode

        create_base_microphone(draw, mic_color, opacity=255)
        create_sound_waves(draw, mic_color, opacity=255, side='left', offset=2)
        create_sound_waves(draw, mic_color, opacity=255, side='right', offset=2)

    # Save the icon
    output_path = os.path.join(output_dir, filename)
    img.save(output_path, 'PNG')
    print(f"Created: {output_path}")

    # Fix reference to mic_top
    mic_height = 18
    mic_top = CENTER - mic_height // 2 - 4

# Generate all tray icons
def main():
    print("Generating tray icons...")

    # Light mode icons
    create_tray_icon('tray_idle.png', 'idle', is_dark=False)
    create_tray_icon('tray_recording.png', 'recording', is_dark=False)
    create_tray_icon('tray_transcribing.png', 'transcribing', is_dark=False)

    # Dark mode icons
    create_tray_icon('tray_idle_dark.png', 'idle', is_dark=True)
    create_tray_icon('tray_recording_dark.png', 'recording', is_dark=True)
    create_tray_icon('tray_transcribing_dark.png', 'transcribing', is_dark=True)

    print("\nAll tray icons generated successfully!")
    print(f"Icons saved to: {output_dir}")

if __name__ == '__main__':
    main()

File: scripts/tools/generate_tray_icons.js
#!/usr/bin/env node
/**
 * Generate tray icons for macOS menu bar
 */

import fs from "fs";
import path from "path";
import { fileURLToPath } from "url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Output directory
const outputDir = path.join(__dirname, "..", "src-tauri", "resources");

// Ensure output directory exists
if (!fs.existsSync(outputDir)) {
  fs.mkdirSync(outputDir, { recursive: true });
}

// SVG templates for each state
const createIdleSVG = (isDark) => `
<svg width="44" height="44" viewBox="0 0 44 44" xmlns="http://www.w3.org/2000/svg">
    <rect x="16" y="12" width="12" height="18" rx="6" fill="${isDark ? "rgba(255,255,255,0.8)" : "rgba(0,0,0,0.7)"}"/>
    <line x1="22" y1="30" x2="22" y2="36" stroke="${isDark ? "rgba(255,255,255,0.8)" : "rgba(0,0,0,0.7)"}" stroke-width="2" stroke-linecap="round"/>
    <line x1="17" y1="36" x2="27" y2="36" stroke="${isDark ? "rgba(255,255,255,0.8)" : "rgba(0,0,0,0.7)"}" stroke-width="2" stroke-linecap="round"/>
</svg>
`;

const createRecordingSVG = (isDark) => `
<svg width="44" height="44" viewBox="0 0 44 44" xmlns="http://www.w3.org/2000/svg">
    <rect x="16" y="12" width="12" height="18" rx="6" fill="${isDark ? "rgb(255,100,100)" : "rgb(220,50,50)"}"/>
    <line x1="22" y1="30" x2="22" y2="36" stroke="${isDark ? "rgb(255,100,100)" : "rgb(220,50,50)"}" stroke-width="2" stroke-linecap="round"/>
    <line x1="17" y1="36" x2="27" y2="36" stroke="${isDark ? "rgb(255,100,100)" : "rgb(220,50,50)"}" stroke-width="2" stroke-linecap="round"/>
    <circle cx="22" cy="7" r="2" fill="${isDark ? "rgb(255,80,80)" : "rgb(255,50,50)"}"/>
    <path d="M 10 17 Q 8 22 10 27" stroke="${isDark ? "rgb(255,100,100)" : "rgb(220,50,50)"}" stroke-width="2" fill="none" stroke-linecap="round" opacity="0.7"/>
    <path d="M 34 17 Q 36 22 34 27" stroke="${isDark ? "rgb(255,100,100)" : "rgb(220,50,50)"}" stroke-width="2" fill="none" stroke-linecap="round" opacity="0.7"/>
</svg>
`;

const createTranscribingSVG = (isDark) => `
<svg width="44" height="44" viewBox="0 0 44 44" xmlns="http://www.w3.org/2000/svg">
    <rect x="16" y="12" width="12" height="18" rx="6" fill="${isDark ? "rgb(167,139,250)" : "rgb(139,92,246)"}"/>
    <line x1="22" y1="30" x2="22" y2="36" stroke="${isDark ? "rgb(167,139,250)" : "rgb(139,92,246)"}" stroke-width="2" stroke-linecap="round"/>
    <line x1="17" y1="36" x2="27" y2="36" stroke="${isDark ? "rgb(167,139,250)" : "rgb(139,92,246)"}" stroke-width="2" stroke-linecap="round"/>
    <path d="M 8 17 Q 6 22 8 27" stroke="${isDark ? "rgb(167,139,250)" : "rgb(139,92,246)"}" stroke-width="2" fill="none" stroke-linecap="round" opacity="0.6"/>
    <path d="M 36 17 Q 38 22 36 27" stroke="${isDark ? "rgb(167,139,250)" : "rgb(139,92,246)"}" stroke-width="2" fill="none" stroke-linecap="round" opacity="0.6"/>
    <path d="M 12 19 Q 10 22 12 25" stroke="${isDark ? "rgb(167,139,250)" : "rgb(139,92,246)"}" stroke-width="1.5" fill="none" stroke-linecap="round" opacity="0.4"/>
    <path d="M 32 19 Q 34 22 32 25" stroke="${isDark ? "rgb(167,139,250)" : "rgb(139,92,246)"}" stroke-width="1.5" fill="none" stroke-linecap="round" opacity="0.4"/>
</svg>
`;

// Function to save SVG (which macOS can render directly)
function saveSVG(filename, svgContent) {
  const outputPath = path.join(outputDir, filename);
  fs.writeFileSync(outputPath, svgContent.trim());
  console.log(`Created: ${outputPath}`);
}

// Main function
function main() {
  console.log("Generating tray icons...\n");

  // Light mode icons
  saveSVG("tray_idle.svg", createIdleSVG(false));
  saveSVG("tray_recording.svg", createRecordingSVG(false));
  saveSVG("tray_transcribing.svg", createTranscribingSVG(false));

  // Dark mode icons
  saveSVG("tray_idle_dark.svg", createIdleSVG(true));
  saveSVG("tray_recording_dark.svg", createRecordingSVG(true));
  saveSVG("tray_transcribing_dark.svg", createTranscribingSVG(true));

  console.log("\nAll tray icons generated successfully!");
  console.log(`Icons saved to: ${outputDir}`);
  console.log(
    "\nNote: SVG icons work on macOS. To convert to PNG, open the HTML file in scripts/tray_icons.html",
  );
}

main();
